python/pyspark/ml/param/__init__.py

```diff
@@ -156,9 +156,16 @@ def hasParam(self, paramName):
         """
         Tests whether this instance contains a param with a given
         (string) name.
+
+        Note: this method is changed in Spark 2.0+ to return `False`
```

Comment:
I would view this one as a bug fix and backport it to branch-1.6. So the note is not necessary.